### PR TITLE
GHA Workflow

### PR DESCRIPTION
**Pull request by mbwhite**
Signed-off-by: Matthew B White <whitemat@uk.ibm.com>

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -18,10 +18,220 @@ on:
...
+          docker login ${DOCKER_REGISTRY_URL} --username=${DOCKER_REGISTRY_USERNAME} --password=${DOCKER_REGISTRY_PASSWORD}
+          echo "Logged in to docker registry"
+          # tag javaenv image to PACKAGE_VERSION
+          docker tag hyperledger/fabric-javaenv hyperledger/fabric-nodeenv:amd64-${{ needs.setup.steps.builddata.outputs.PACKAGE_VERSION }}
+          # push javaenv to repository
+          docker push hyperledger/fabric-javaenv:amd64-${{ needs.setup.steps.builddata.outputs.PACKAGE_VERSION }}
+          ./manifest-tool push from-args --platforms linux/amd64 --template "hyperledger/fabric-nodeenv:amd64-${{ needs.setup.steps.builddata.outputs.PACKAGE_VERSION }}" --target "hyperledger/fabric-nodeenv:${{ needs.setup.steps.builddata.outputs.PACKAGE_VERSION }}"
+          ./manifest-tool push from-args --platforms linux/amd64 --template "hyperledger/fabric-nodeenv:amd64-${{ needs.setup.steps.builddata.outputs.PACKAGE_VERSION }}" --target "hyperledger/fabric-nodeenv:${{ needs.setup.steps.builddata.outputs.MINOR_PACKAGE_VERSION }}"
+        env:
+          DOCKER_REGISTRY_USERNAME:  ${{ secrets.dockerhub-username }}
```
Comment (ryjones) [inline .github/workflows/build.yaml:236]: DOCKERHUB_USERNAME

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -18,10 +18,220 @@ on:
...
+          DOCKER_REGISTRY_PASSWORD:  ${{ secrets.dockerhub-password }}
```
Comment (ryjones) [inline .github/workflows/build.yaml:237]: DOCKERHUB_TOKEN

**Review by ryjones — DISMISSED** [review]
two small changes plus similar from JT-NTI's review on the other

**Review by jt-nti — APPROVED** [review]
Nothing uses BUILD_DATE but looks like a reasonable initial workflow

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -18,10 +18,220 @@ on:
...
+      - uses: actions/download-artifact@v3
+        with:
+          name: node-tgzs
+          path: build/          
+      - run: |
+          set -xev
+          ls -lart build/
+          cd build
+          find . -type f -name 'fabric-*.tgz' -exec npm publish {} \;
+        env:
```
[the remark this answers is not in the document]
Comment (jt-nti) [reply .github/workflows/build.yaml:199]: I tend to put it after the run stanza, which should be fine, e.g.
https://github.com/hyperledger-labs/fabric-builder-k8s/blob/main/.github/workflows/go.yml#L64

**Pull request merged**